cmd/skipper/filterFlags.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"strings"
+)
+
+const filterPluginUsage = "set a custom filter plugins to load, a comma separated list of name and arguments"
+
+type filterFlags struct {
+	values [][]string
+}
+
+func (f *filterFlags) String() string {
+	var ret []string
+	for _, val := range f.values {
+		ret = append(ret, strings.Join(val, ","))
+	}
+	return strings.Join(ret, " ")
+}
+
+func (f *filterFlags) Set(value string) error {
+	if f == nil {
+		f = &filterFlags{}
+	}
+	for _, v := range strings.Split(value, " ") {
+		f.values = append(f.values, strings.Split(v, ","))
+	}
+	return nil
+}
+
+func (f *filterFlags) Get() [][]string {
+	return f.values
+}
+
+const predicatePluginUsage = "set a custom predicate plugins to load, a comma separated list of name and arguments"
+
+type predicateFlags struct {
+	values [][]string
+}
+
+func (f *predicateFlags) String() string {
+	var ret []string
+	for _, val := range f.values {
+		ret = append(ret, strings.Join(val, ","))
+	}
+	return strings.Join(ret, " ")
+}
+
+func (f *predicateFlags) Set(value string) error {
+	if f == nil {
+		f = &predicateFlags{}
+	}
+	for _, v := range strings.Split(value, " ") {
+		f.values = append(f.values, strings.Split(v, ","))
+	}
+	return nil
+}
+
+func (f *predicateFlags) Get() [][]string {
+	return f.values
+}
+
+const dataclientPluginUsage = "set a custom dataclient plugins to load, a comma separated list of name and arguments"
+
+type dataclientFlags struct {
+	values [][]string
+}
+
+func (f *dataclientFlags) String() string {
+	var ret []string
+	for _, val := range f.values {
+		ret = append(ret, strings.Join(val, ","))
+	}
+	return strings.Join(ret, " ")
+}
+
+func (f *dataclientFlags) Set(value string) error {
+	if f == nil {
+		f = &dataclientFlags{}
```

Comment:
i don't think this works. We're not setting here a value for the *f, but we're creating this way a new pointer, replacing f. This way the calling code will still have the old nil value. https://play.golang.org/p/ttasuPgILeI

if we were setting *f, that still would not be good, because it's nil, so we cannot dereference it.

I would simple remove this nil check and expect the calling code to initialize the pointer as usual.